secure/key-manager/src/lib.rs

```diff
@@ -108,9 +114,33 @@ where
             libra,
             storage,
             time_service,
+            last_checked_libra_timestamp: 0,
         }
     }
 
+    /// Begins execution of the key manager by running an infinite loop where the key manager will
+    /// periodically wake up, verify the state of the validator keys (e.g., the consensus key), and
+    /// initiate a key rotation when required. If something goes wrong, an error will be returned
+    /// by this method, upon which the key manager will flag the error and stop execution.
+    pub fn execute(&mut self) -> Result<(), Error> {
+        loop {
+            self.execute_once_then_sleep()?;
+        }
+    }
+
+    /// Checks the current state of the validator keys and performs any actions that might be
+    /// required (e.g., performing a key rotation). Once all checks and steps have been completed,
+    /// sleep until the next iteration needs to be performed.
+    pub fn execute_once_then_sleep(&mut self) -> Result<(), Error> {
+        let status = self.evaluate_status()?;
+        if status != Action::NoAction {
+            self.perform_action(status)?;
+        }
+
+        self.time_service.sleep(SLEEP_PERIOD_SECS);
```

Comment:
let's put this into the execute statement, for testing purposes having an execute_once seems handy

Reply:
Done, thanks!